core_impl/src/main/java/io/opencensus/implcore/tags/propagation/SerializationUtils.java

```diff
@@ -152,6 +151,26 @@ static TagContextImpl deserializeBinary(byte[] bytes) throws TagContextParseExce
     }
   }
 
+  // TODO(sebright): Consider exposing a TagKey name validation method to avoid needing to catch an
+  // IllegalArgumentException here.
+  private static final TagKeyString createTagKey(String name) throws TagContextParseException {
+    try {
+      return TagKeyString.create(name);
+    } catch (IllegalArgumentException e) {
+      throw new TagContextParseException("Invalid tag key: " + name, e);
```

Comment:
Is this just to have a better exception message? You can simply catch IllegalArgumentException in the deserializeBinary and avoid this 2 methods.

Reply:
I was trying to give a better error message, so that we don't need to rely on the IllegalArgumentException giving the user all of the information that they need.  I also thought that we might want to add something like `TagKey.isValidName(String)` later, if throwing the IllegalArgumentException is inefficient, which would also require the TagContextParseException to be thrown where the TagKey is created.

Reply:
I would not use TagContextParseException for TagKey creation because it is not a parsing exception in my opinion.

Reply:
What do you think we should do?  Should we use a different custom checked exception (possibly sharing a superclass with TagContextParseException)?  I'd also like to document the error handling for invalid tags in https://github.com/census-instrumentation/opencensus-specs, since this should be consistent across languages.

Reply:
For the moment we are ok we do not change the API so we are good. Please document this.

Reply:
I created an issue for improving the handling of invalid tags: #674

Reply:
I also updated fromByteArray's Javadoc: #675